hook!-full-game/src/Player/States/Hook.gd

```diff
@@ -16,7 +16,7 @@ func physics_process(delta: float) -> void:
 	
 	self.velocity = Steering.arrive_to(velocity, owner.global_position, target_global_position, HOOK_MAX_SPEED)
 	self.velocity = owner.move_and_slide(velocity, owner.FLOOR_NORMAL)
-	if distance < velocity.length() * delta:
+	if distance < velocity.length() * delta or owner.is_on_ceiling():
```

Comment:
This is happening because you didn't separate your branches properly.

It's a bit tricky with git. To split changes in 2 PRs:

1. Create a first branch from Master
2. Check out to the new branch
3. Create your first commit there. you do not need to stash your changes first when you do so.
4. (optionally) Checkout back to master
5. Create a new branch that starts from Master
6. Create your second commit with the remaining changes there

Then push and create the two pull requests like usual